### PR TITLE
Use argparse instead of own argument parsing

### PR DESCRIPTION
**Pull request by lmm-git**
and allow to specify the latex engine by command line, not only environment variable

### CODE REVIEW

**Comment by TimNN** [comment]
Have you tested this?

File: lbuild [hunk 1]
```diff
@@ -22,21 +23,22 @@ def main():
     sys.exit(0)
 
 def real_main():
-    assert len(sys.argv) == 2
+    parser = argparse.ArgumentParser(description='Building LaTeX documents')
+    parser.add_argument('texpath', type=str, help='Path to tex file')
+    parser.add_argument('--latexEngine', type=str, help='LaTeX engine to use. Possible engines are "lualatex", "pdflatex", "xelatex".',
```
Comment (TimNN) [inline lbuild]: Could you rename the flag to '--latex-engine'?

**Comment by lmm-git** [comment]
yes

File: lbuild [hunk 2]
```diff
@@ -51,7 +54,11 @@ def process(tex):
 
         print("Now compiling {}".format(tex))
 
-        ret = subprocess.run(cmd, timeout=300, stdin=DEVNULL, stdout=DEVNULL, stderr=DEVNULL)
+        try:
+            ret = subprocess.run(cmd, timeout=timeout, stdin=DEVNULL, stdout=DEVNULL, stderr=DEVNULL)
+        except subprocess.TimeoutExpired:
+            print("Timeout exceeded!")
```
Comment (TimNN) [inline lbuild:60]: Why add that print?

Comment (lmm-git) [inline lbuild:60]: Because you catch all exceptions in the main function and I wanted to present the user some kind of error message

**Pull request merged**